lib/fizzy/parser_expr.cpp

```diff
@@ -111,6 +119,18 @@ void validate_result_count(const ControlFrame& frame)
     //     throw validation_error{"too many results"};
 }
 
+void push_branch_immediates(const ControlFrame& frame, bytes& immediates)
+{
+    // Push frame start location as br immediates - these are final if frame is loop,
+    // but for block/if/else these are just placeholders, to be filled at end instruction.
+    push(immediates, static_cast<uint32_t>(frame.code_offset));
+    push(immediates, static_cast<uint32_t>(frame.immediates_offset));
+    push(immediates, static_cast<uint32_t>(frame.parent_stack_height));
+    // Always pushing 0 as loop's arity, because br executed in loop jumps to the top,
+    // resetting frame stack to 0, so return type arity is not important for br resolution.
+    push(immediates, frame.instruction == Instr::loop ? uint8_t{0} : frame.arity);
```

Comment:
If arity for loop is ignored, couldn't we just always pass `frame.arity` to avoid this extra branch? Likely it is an unneeded micro-optimization at this stage.

Reply:
No, if loop has a result, it still needs to clear all the stack when executing `br`, not keeping one element like `br` out of `block` or `if`.

This way it's less branching in execution - branching code doesn't care from what kind of block its jumping, and just takes destination and arity from immediates.

Reply:
Rephrased the comment slightly.

Reply:
This comment wording to me still suggests that loop always gets arity 0, just like the code does.